server/src/main/java/org/apache/druid/segment/realtime/plumber/Sink.java

```diff
@@ -82,7 +88,12 @@ public class Sink implements Iterable<FireHydrant>, Overshadowable<Sink>
   private final long maxBytesInMemory;
   private final boolean useMaxMemoryEstimates;
   private final CopyOnWriteArrayList<FireHydrant> hydrants = new CopyOnWriteArrayList<>();
+
   private final LinkedHashSet<String> dimOrder = new LinkedHashSet<>();
+  // columns excluding current index, includes __time column
```

Comment:
what is curIndex, the inmemory Fire hydrant ? If yes lets rename it like that / or document it